aiida/cmdline/commands/cmd_rehash.py

```diff
@@ -29,14 +29,35 @@
     default=None,
     help='Only include nodes that are class or sub class of the class identified by this entry point.'
 )
+@options.FORCE()
 @decorators.with_dbenv()
-def rehash(nodes, entry_point):
-    """Recompute the hash for nodes in the database
+def rehash(nodes, entry_point, force):
+    """Recompute the hash for nodes in the database.
 
     The set of nodes that will be rehashed can be filtered by their identifier and/or based on their class.
     """
     from aiida.orm import Data, ProcessNode, QueryBuilder
 
+    if not force:
+        echo.echo_warning('This command will recompute and overwrite the hashes of all nodes.')
+        echo.echo_warning('Note that this can take a lot of time for big databases.')
+        echo.echo_warning('Also, do it only if you are sure of what you are doing.')
+        echo.echo_warning('')
+        echo.echo_warning('', nl=False)
+
+        confirm_message = 'Do you want to continue? [y/n]'
+
+        try:
+            while True:
```

Comment:
why the while loop? We can do just a single if statement:
```
if not click.prompt():
    echo.echo_critical()
```

Reply:
It's in case the user replies with a different letter. Do you want to stop also in that case?

Reply:
Ok, I changed to use click.prompt's list of valid choices